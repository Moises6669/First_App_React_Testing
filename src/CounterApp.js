import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  //handleAdd
  const handleAdd = () => setCounter(counter + 1);

  const handlessOne = () => setCounter(counter - 1);

  const handlreset = () => setCounter(value);
  
  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}> +1 </button>
      <button onClick={handlessOne}> -1 </button>
      <button className="button_reset" onClick={handlreset}>reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  value: 1,
};

export default CounterApp;
