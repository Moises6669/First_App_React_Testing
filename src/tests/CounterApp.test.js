import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Analizando valores del componente <CounterApp />", () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("Validando el valor por defecto", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Mofidicando el valor por defecto", () => {
    const wrapper = shallow(<CounterApp value={100} />);
    const h2num = wrapper.find("h2").text().trim();
    expect(h2num).toBe("100");
  });

  test("debe de incrementar con el boton de +1", () => {
    wrapper.find("button").at(0).simulate("click");
    const h2num = wrapper.find("h2").text().trim();
    expect(h2num).toBe("2");
  });

  test("debe de incrementar con el boton de -1", () => {
    wrapper.find("button").at(1).simulate("click");
    const h2num = wrapper.find("h2").text().trim();
    expect(h2num).toBe("0");
  });

  test("debe de colocar el valor por defecto con el boton de reset", () => {
    const wrapper = shallow(<CounterApp value={105} />);
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find(".button_reset").simulate("click");
    const buttonPlus = wrapper.find("h2").text().trim();
    expect(buttonPlus).toBe("105");
  });
});
