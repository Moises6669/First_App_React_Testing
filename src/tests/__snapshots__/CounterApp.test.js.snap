// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analizando valores del componente <CounterApp /> Validando el valor por defecto 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    1
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
  <button
    className="button_reset"
    onClick={[Function]}
  >
    reset
  </button>
</Fragment>
`;
